src/ILVerify/src/ILImporter.StackValue.cs

```diff
@@ -362,7 +362,7 @@ bool IsBinaryComparable(StackValue src, StackValue dst, ILOpcode op)
                         case StackValueKind.ObjRef:
                             return op == ILOpcode.beq || op == ILOpcode.beq_s ||
                                    op == ILOpcode.bne_un || op == ILOpcode.bne_un_s ||
-                                   op == ILOpcode.ceq;
+                                   op == ILOpcode.ceq || op == ILOpcode.cgt_un;
```

Comment:
It may be useful to quote the one sentence that this is allowed from the ECMA spec for the `cgt_un/bgt_un` cases. It was a surprise to me - I did not know that this corner case exists.